app/models/concerns/waste_carriers_engine/can_change_registration_status.rb

```diff
@@ -100,9 +99,24 @@ def should_not_be_expired?
       expiry_day = expiry_time_adjusted_for_daylight_savings.to_date
       # We store dates and times in UTC, but want to use the current date in the UK, not necessarily UTC
       current_day = Time.now.in_time_zone("London").to_date
+
+      return true if in_expiry_grace_window?(expiry_day, current_day)
```

Comment:
I think this addition changes the meaning of the method and makes the name inaccurate. This method was initially intended to check whether the registration's status should be 'EXPIRED' and would be changed as soon as the scheduled job hit it, thus the concept of "should be expired".

However, now it could still return `true` if it should be expired _but_ happens to be in the grace window. This feels inaccurate to me. I think we are conflating the concepts of a registration being expired and not being renewable, but they're not the same thing.

So I think it would be better to call `in_expiry_grace_window?` from `renewal_allowed?` instead of putting it in this method.

This may mean a bit of additional refactoring to reduce duplication of local `expiry_day` variables (which I notice are inconsistently calculated anyway - a pre-existing problem though!) but I'm OK with that. What do you think?